src/DittoMachine.sol

```diff
@@ -485,16 +485,19 @@ contract DittoMachine is ERC721, ERC721TokenReceiver, ERC1155TokenReceiver, Clon
         // token can only be sold to the clone at the index head
         popListHead(protoId);
 
+        // send useful data along with safe transfer to sontracts
+        bytes memory data = abi.encode(cloneId, owner, cloneShape.worth, subsidy);
```

Comment:
should we also send `protoId`?

Reply:
I'm wondering if maybe we should send all of the data needed to calculate both the `protoId` and `cloneId` and let the receiving contract calculate them if they need it. what do you think?

Reply:
Also now that i didn't need to include owner in the data because they will be receiving the function call anyway

Reply:
In that case, I think the receiving contract already knows all the details to calculate `protoId` and `cloneId`

Reply:
`protoIdToIndexHead` will have already moved on to the next index. Additionally I believe sending the data with the function call would save gas by reducing storage reads

Reply:
sounds good. then let's do what you said above.